controllers/operatorconfig_controller.go

```diff
@@ -54,51 +78,315 @@ type OperatorConfigReconciler struct {
+	if err != nil {
+		return err
+	}
+
+	// Add Finalizer for this CR
+	if !controllerutil.ContainsFinalizer(&obj, Finalizer) {
+		controllerutil.AddFinalizer(&obj, Finalizer)
+	}
+
+	obj.Labels = map[string]string{
+		OwnerLabel: OwnerConfigName,
+	}
+
+	if err := r.Create(ctx, &obj); err != nil {
+		if !errors.IsAlreadyExists(err) {
+			log.Error(err, fmt.Sprintf("Failed to create %s", obj.Kind), "Namespace", obj.Namespace, "Name", obj.Name)
+			return err
+		}
+	} else {
+		log.Info(fmt.Sprintf("Created %s/%s", obj.Kind, obj.Name))
+	}
+
+	c := pipelinev1beta1.ClusterTask{}
+	if err := r.Get(ctx, types.NamespacedName{Namespace: obj.Namespace, Name: obj.Name}, &c); err != nil {
+		return err
+	}
+
+	if !reflect.DeepEqual(c.Spec, obj.Spec) {
+		c.Spec = obj.Spec
+		if err := r.Update(ctx, &c); err != nil {
+			log.Error(err, fmt.Sprintf("Failed to update %s", obj.Kind), "Namespace", obj.Namespace, "Name", obj.Name)
+			return err
+		}
+	}
+
+	return nil
 }
 
-func configureCranePlugin(ctx context.Context, log logr.Logger) error {
-	return createResourcesFromFile("crane-ui-plugin.yaml", ctx, log)
+func (r *OperatorConfigReconciler) reconcileConsolePlugin(resource *unstructured.Unstructured, ctx context.Context, log logr.Logger, oc *cranev1alpha1.OperatorConfig) error {
+	var obj consolev1alpha1.ConsolePlugin
+	err := runtime.DefaultUnstructuredConverter.
+		FromUnstructured(resource.UnstructuredContent(), &obj)
+	if err != nil {
+		return err
+	}
+
+	obj.Labels = map[string]string{
+		OwnerLabel: OwnerConfigName,
+	}
+
+	if err := r.Create(ctx, &obj); err != nil {
+		if !errors.IsAlreadyExists(err) {
+			log.Error(err, fmt.Sprintf("Failed to create %s", obj.Kind), "Namespace", obj.Namespace, "Name", obj.Name)
+			return err
+		}
+	} else {
+		log.Info(fmt.Sprintf("Created %s/%s", obj.Kind, obj.Name))
+	}
+
+	c := consolev1alpha1.ConsolePlugin{}
+	if err := r.Get(ctx, types.NamespacedName{Namespace: obj.Namespace, Name: obj.Name}, &c); err != nil {
+		return err
+	}
+
+	if !reflect.DeepEqual(c.Spec, obj.Spec) {
+		c.Spec = obj.Spec
+		if err := r.Update(ctx, &c); err != nil {
+			log.Error(err, fmt.Sprintf("Failed to update %s", obj.Kind), "Namespace", obj.Namespace, "Name", obj.Name)
+			return err
+		}
+	}
+
+	return nil
 }
 
-func createClusterTasks(ctx context.Context, log logr.Logger) error {
-	return createResourcesFromFile("manifests.yaml", ctx, log)
+func (r *OperatorConfigReconciler) deleteResourcesFromFile(path string, ctx context.Context) error {
+	var decoder = yaml.NewDecodingSerializer(unstructured.UnstructuredJSONScheme)
+	data, err := getResources(path)
+	if err != nil {
+		return err
+	}
+
+	for _, resource := range data {
+		if len(resource) > 0 {
+			obj := &unstructured.Unstructured{}
+			_, _, err := decoder.Decode([]byte(resource), nil, obj)
+			if err != nil {
+				return err
+			}
+
+			if err = r.Get(ctx, types.NamespacedName{Name: obj.GetName(), Namespace: obj.GetNamespace()}, obj); err != nil {
+				return err
```

Comment:
do we need to check for `NotFound` error before returning err? I assume if not found, we just continue to look through